src/chain.rs

```diff
@@ -0,0 +1,56 @@
+//! Chain specific utils
+
+use std::{collections::HashMap, fmt::Debug};
+
+use alloy_primitives::U256;
+use alloy_rpc_types::{Header, Transaction};
+use revm::primitives::{BlockEnv, SpecId, TxEnv};
+
+use crate::{
+    mv_memory::{LazyAddresses, MvMemory},
+    BuildIdentityHasher, MemoryLocation, TxIdx,
+};
+
+/// Custom behaviours for different chains & networks
+pub trait PevmChain: Debug + Clone + PartialEq {
```

Comment:
Do we really need these `Debug + Clone + PartialEq`? If not then let's not restrict it. We never know if the actual struct implementing the trait would be easy/hard to compare or clone.

Reply:
It seems that `Clone` is redundant but currently we need PartialEq so that we can compare two `PevmResult` values:

```
pub fn assert_execution_result<C: PevmChain>(
    sequential_result: &PevmResult<C>,
    parallel_result: &PevmResult<C>,
) {
    assert_eq!(sequential_result, parallel_result);
}
```

Should we compare only the `Ok` arms instead, and ignore if both `sequential_result` and and `parallel_result` are `Err`?

Reply:
The most natural way is to restrict at `assert_execution_result` with something like:
```
pub fn assert_execution_result<C: PevmChain + PartialEq>(...
```

Meaning we can only assert execution results of chain objects that can be compared, but not all chain objects must be able to be compared.

Ideally, we wouldn't even need to require it as only the error types are represented in `PevmResults`, and it's fair to enforce these traits on error types.

Reply:
The change is bigger than expected because we must implement `impl<C> PartialEq for PevmError<C>` manually. `#[derive(PartialEq)]` does not work. Please help to review commit https://github.com/risechain/pevm/pull/230/commits/a00024de6d1a72a9c9dc756e722bc3dad54be59e

Reply:
I updated the code. I only removed the trait bounds for `PevmChain`. You can compare the previous solution (https://github.com/risechain/pevm/commit/a00024de6d1a72a9c9dc756e722bc3dad54be59e) and the current solution (https://github.com/risechain/pevm/pull/230/commits/84aca8672c6cd81ed5eaada851c4d13566a725ba).